x-pack/filebeat/input/awscloudwatch/cloudwatch.go

```diff
@@ -151,14 +154,30 @@ func (p *cloudwatchPoller) startWorkers(
 // equal time.Now) to allow deterministic unit tests.
 func (p *cloudwatchPoller) receive(ctx context.Context, logGroupIDs []string, clock func() time.Time) {
 	defer p.workerWg.Wait()
+
 	// startTime and endTime are the bounds of the current scanning interval.
-	// If we're starting at the end of the logs, advance the start time to the
-	// most recent scan window
-	var startTime time.Time
 	endTime := clock().Add(-p.config.Latency)
-	if p.config.StartPosition == "end" {
+
+	var startTime time.Time
+	// If we're starting at the end of the logs, advance the start time to the most recent scan window
+	if p.config.StartPosition == End {
 		startTime = endTime.Add(-p.config.ScanFrequency)
 	}
+
+	if p.config.StartPosition == Beginning {
+		startTime = time.Unix(0, 0)
+	}
+
+	if p.config.StartPosition == LastSync {
+		state, err := p.stateHandler.GetState(p.config)
+		if err != nil {
+			p.log.Errorf("error retriving state from stateHandler: %v, falling back to %s", err, Beginning)
```

Comment:
```suggestion
			p.log.Errorf("error retrieving state from stateHandler: %v, falling back to %s", err, Beginning)
```

Reply:
Thank you and done - https://github.com/elastic/beats/pull/43251/commits/86c216643f6c72b7eb6c6f536db45d100777b67f